homeassistant/components/synology_dsm/__init__.py

```diff
@@ -149,7 +154,7 @@ def _async_migrator(entity_entry: entity_registry.RegistryEntry):
         ):
             return None
 
-        entity_type = None
+        entity_type = ""
```

Comment:
Return some error if it's `None` after the loop.
```suggestion
        entity_type: str | None = None
```